File: Problem_1.py
import java.util.Scanner;

public class UserLogin {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        String username = "";
        String password = "";
        String email = "";
       
        while (true) {
            System.out.print("Username: ");
            username = scanner.nextLine().trim();

            if (username.length() > 50 || username.isEmpty() ) {
                System.out.println("Username is invalid");
            }
            else {
                System.out.println("Username is valid");
            }
       
       
        System.out.print("Password: ");
        password = scanner.nextLine().trim();
       
        if (password.length() > 50) {
                System.out.println("Password must be atleast 8 characters long, Please try again");
            }  
       
       String regex = "^(?=.*[a-zA-Z])(?=.*\\d)(?=.*[A-Z])(?=.*[a-z]).{8,}$";

        if (password.matches(regex)) {
            System.out.println("Password is valid.");
        } else {
            System.out.println("Password is invalid");
        }
       
        System.out.print("Email: ");
        email = scanner.nextLine().trim();
           
        String regex2 = "^[a-zA-Z0-9]+@[a-zA-Z0-9]+\\.[a-zA-Z]+$";

        if (email.matches(regex2)) {
            System.out.println("Email is valid.");
        } else {
            System.out.println("Email is invalid");
        }
break;
    }
           
       
    }
}

File: problem_1.java
import java.util.Scanner;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class ULV {
    public static void main(String[] args) {
        Scanner s = new Scanner(System.in);
            System.out.println("Username:");
            String usrnm = s.nextLine();
            System.out.println("Password:");
            String psswrd = s.nextLine();
            System.out.println("Email:");
            String eml = s.nextLine();

            boolean isValdU = isValdU(usrnm);
            boolean isValdP = isValdP(psswrd);
            boolean isValdE = isValdE(eml);

            if (!isValdU) {
                System.out.println("Username is invalid.");
            } else {
                System.out.println("Username is valid.");
            }
            if (!isValdP) {
                System.out.println("Password is invalid.");
            } else {
                System.out.println("Password is valid.");
            }
            if (!isValdE) {
                System.out.println("Email is invalid.");
            } else {
                System.out.println("Email is valid.");
            }


    }

    static boolean isValdU(String usrnm) {
        if (usrnm.isEmpty() || usrnm.length() > 50) {
            return false;
        }
        return true;
    }

    static boolean isValdP(String psswrd) {
        if (psswrd.length() < 8 || !psswrd.matches(".*[a-z].*") || !psswrd.matches(".*[A-Z].*") || !psswrd.matches(".*\\d.*") || !psswrd.matches(".*[!@#$%^&*()].*")) {
            return false;
        }
        return true;
    }

    static boolean isValdE(String eml) {
        String regex = "^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,}$";
        Pattern p = Pattern.compile(regex);
        Matcher m = p.matcher(eml);
        return m.matches();
    }
}

